worker/worker.go

```diff
@@ -966,7 +967,15 @@ func (w *Worker) HeadObject(ctx context.Context, bucket, key string, opts api.He
 	return res, err
 }
 
-func (w *Worker) SyncAccount(ctx context.Context, fcid types.FileContractID, hk types.PublicKey, siamuxAddr string) error {
+func (w *Worker) SyncAccount(ctx context.Context, fcid types.FileContractID, host api.HostInfo) error {
+	// handle v2 host
+	if host.IsV2() {
+		account := w.accounts.ForHost(host.PublicKey)
+		return account.WithSync(func() (types.Currency, error) {
+			return w.rhp4Client.AccountBalance(ctx, host.PublicKey, host.V2SiamuxAddr(), rhpv4.Account(account.ID()))
```

Comment:
Is this free? Asking because there's probably some loose ends with regards to gouging. For v1 hosts we have to attach a gouging checker because we perform gouging checks on the price table when we fetch it. That's something I think we should stop doing though. Anway with RHP4 all of this will change anyway but wanted to remind you in case you missed the `WithGougingChecker`

Reply:
Yes, it is free in RHP4